pkgs/os-specific/linux/procps-ng/default.nix

```diff
@@ -1,4 +1,10 @@
-{ lib, stdenv, fetchurl, ncurses, pkgconfig }:
+{ lib, stdenv, fetchurl, ncurses, pkgconfig
+
+# procps is mostly Linux-only. Most commands require a running Linux
+# system (or very similar like that found in Cygwin). The one
+# exception is ‘watch’ which is portable enough to run on pretty much
+# any UNIX-compatible system.
+, watchOnly ? !(stdenv.isLinux || stdenv.isCygwin) }:
```

Comment:
`stdenv.isDarwin` seems more precise

Reply:
It should work on BSD systems as well at least: http://pkgsrc.se/wip/watch

Reply:
The correct fix for this mess would be for someone to step up and maintain a fork of procps with just watch enabled. But apparently no one has done that and I don't think anyone else wants to.

Reply:
Another approach might be to have to version of procps-ng: watch only and everything but watch. That would reduce the amount of condition switches.

Reply:
Yes that  would work but might be a little harder to maintain. The old watch derivation kind of worked like this the only problem was that it was pretty easy for it to become out of date - procps-ng was the only one updated. At least now CI we will have watch built as part of ```procps```.